lib/src/combine_latest.dart

```diff
@@ -174,7 +174,7 @@ extension CombineLatest<T> on Stream<T> {
   /// If the source stream is a broadcast stream, the result stream will be as
   /// well, regardless of the types of [others]. If a single subscription stream
   /// is combined with a broadcast source stream, it may never be canceled.
-  Stream<List<T>> combineLatestAll(Iterable<Stream<T>> others) {
+  Stream<List<T>> combineLatestAll(Iterable<Stream<T>> others, {bool waitAll = true}) {
```

Comment:
We will need to update the Doc comment and add a CHANGELOG entry

Reply:
@lrhn - any thoughts on the argument name? If I was writing the PR I'd probably go with `emitPartial`.

This is likely a rarely used argument, so a long name should be fine.

Reply:
> But I don't like the design to begin with, so I'd go a different way than a boolean flag.
> 
> Removing the `null` values removes the correspondence between value and source position.

I think this is a caveat that we could document - it only impacts the partial lists emitted before all inputs have emitted a value.

Reply:
The only way to know if a list is partial is knowing the number of streams in the original iterable.

I just don't think it's a good design to mix the partial and complete results in the same function. The two exposed behaviors are too different, even if the events are found by the same underlying logic.

So: One function for complete lists, non -nullable element types, one or two which also include partial lists, one can be nullable, the other can remove the nulls and not preserve position, but the latter is easily computer from the former, so if probably go with only one.

It doesn't have to be the same function. Boolean flags which change behavior, in this case straight away from the original design, may be better off as a separate function.